packagehandlers/nugetpackagehandler.go

```diff
@@ -23,5 +36,90 @@ func (nph *NugetPackageHandler) UpdateDependency(vulnDetails *utils.Vulnerabilit
 }
 
 func (nph *NugetPackageHandler) updateDirectDependency(vulnDetails *utils.VulnerabilityDetails) (err error) {
-	return nph.CommonPackageHandler.UpdateDependency(vulnDetails, vulnDetails.Technology.GetPackageInstallationCommand(), dotnetPackageUpgradeExtraArg)
+	var assetsFilePaths []string
+	assetsFilePaths, err = getAssetsFilesPaths()
+	if err != nil {
+		return
+	}
+
+	wd, err := os.Getwd()
+	if err != nil {
+		return
+	}
+
+	vulnRegexpCompiler := getVulnerabilityRegexCompiler(vulnDetails.ImpactedDependencyName, vulnDetails.ImpactedDependencyVersion)
+	var isAnyFileChanged bool
+
+	for _, assetFilePath := range assetsFilePaths {
+		var isFileChanged bool
+		isFileChanged, err = fixNugetVulnerabilityIfExists(nph, vulnDetails, assetFilePath, vulnRegexpCompiler, wd)
+		if err != nil {
+			err = fmt.Errorf("failed to update asset file '%s': %s", assetFilePath, err.Error())
+			return
+		}
+
+		// We use logic OR in order to keep track whether any asset file changed during the fix process
+		isAnyFileChanged = isAnyFileChanged || isFileChanged
+	}
+
+	if !isAnyFileChanged {
+		err = fmt.Errorf("impacted package '%s' was not found or could not be fixed in all descriptor files", vulnDetails.ImpactedDependencyName)
+	}
+	return
+}
+
+func getAssetsFilesPaths() (assetsFilePaths []string, err error) {
+	err = filepath.WalkDir(".", func(path string, d fs.DirEntry, innerErr error) error {
+		if innerErr != nil {
+			return fmt.Errorf("error has occurrd when trying to access or traverse the files system: %s", innerErr.Error())
+		}
+
+		if strings.HasSuffix(path, dotnetAssetsFilesSuffix) {
+			var absFilePath string
+			absFilePath, innerErr = filepath.Abs(path)
+			if innerErr != nil {
+				return fmt.Errorf("couldn't retrieve file's absolute path for './%s': %s", path, innerErr.Error())
+			}
+			assetsFilePaths = append(assetsFilePaths, absFilePath)
+		}
+		return nil
+	})
+	return
+}
+
+func fixNugetVulnerabilityIfExists(nph *NugetPackageHandler, vulnDetails *utils.VulnerabilityDetails, assetFilePath string, vulnRegexpCompiler *regexp.Regexp, originalWd string) (isFileChanged bool, err error) {
+	modulePath := path.Dir(assetFilePath)
+
+	var fileData []byte
+	fileData, err = os.ReadFile(assetFilePath)
+	if err != nil {
```

Comment:
Let's wrap this error with informative context, so that if something fails we will have a context for debugging.
Let's do the same for all `os` functions.